pulpcore/app/viewsets/publication.py

```diff
@@ -264,6 +264,9 @@ class DistributionViewSet(
     serializer_class = DistributionSerializer
     filterset_class = DistributionFilter
 
-    def async_reserved_resources(self, instance):
+    def async_reserved_resources(self, instance, request):
         """Return resource that locks all Distributions."""
-        return ["/api/v3/distributions/"]
+        if "base_path" in request.data:
+            return ["/api/v3/distributions/"]
```

Comment:
```suggestion
            return [instance, "/api/v3/distributions/"]
```
If we treat `"/api/v3/distributions/"` as the `base_bath` lock, we need to grab the one for the instance anyway.